internal/mode/static/handler.go

```diff
@@ -92,3 +118,81 @@ func (h *eventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Confi
 
 	return nil
 }
+
+// updateControlPlane updates the control plane configuration with the given user spec.
+// If any fields are not set within the user spec, the default configuration values are used.
+func (h *eventHandlerImpl) updateControlPlane(ctx context.Context, cfg *nkgAPI.NginxGateway) {
+	// build up default configuration
+	defaultLogLevel := nkgAPI.ControllerLogLevelInfo
+	controlConfig := nkgAPI.NginxGatewaySpec{
+		Logging: &nkgAPI.Logging{
+			Level: &defaultLogLevel,
+		},
+	}
+
+	updateControlPlane := func() error {
+		// by marshaling the user config and then unmarshaling on top of the default config,
+		// we ensure that any unset user values are set with the default values
```

Comment:
Does json.Unmarshal understand `kubebuilder` annotations? I wonder how it can get the default values

Reply:
We define the default values above in a struct literal.